File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int mxjump = 0;

        for(int i =0 ;i < nums.size(); i++) {
            if(i > mxjump) return false;
            mxjump = max(mxjump, i+nums[i]);
        }
        return true;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        pref = [prices[0]]
        n = len(prices)
        for i in range(1,n):
            pref.append(min(pref[-1],prices[i]))
        
        mx = -1
        for i in range(1,n):
            mx=max(mx,prices[i]-pref[i-1])
        return max(mx,0)

        